src/KnownProperties.jl

```diff
@@ -0,0 +1,67 @@
+########################################################################
+# This is to introduce a common language to ask whether or not a given
+# property (which is potentially hard to compute) is known for some
+# object `x`. 
+########################################################################
+
+@doc raw"""
+    is_known(x::Any, f::Function)
+
+Return whether a property of an object `x` called for by `f(x)` is known. 
+
+Note: The default implementation throws an error. It is the programmer's 
+responsibility to implement appropriate methods for their individual 
+types and properties. See `src/KnownProperties.jl` for details.
+"""
+function is_known(x::Any, f::Function)
+  # If the object `x` has attributes and an attribute with the name 
+  # of `f` happens to be stored, return it in good faith.
+  hasfield(typeof(x), :__attrs) && has_attribute(x, nameof(f)) && return true
```

Comment:
```suggestion
  _is_attribute_storing_type(typeof(x)) && has_attribute(x, nameof(f)) && return true
```
for singleton types, there is no field `__attrs` even though they support attributes

Reply:
Thanks, good to know.